### DISCUSSION TITLE
howto pin a buffer through sessions switching?

### DISCUSSION THREAD

**Discussion by tmpm697** [olimorris/persisted.nvim#82] [General]
### Your Persisted.nvim config
```
  {
    name = "PersistedHooks",
    {
      "User",
      function(session)
        require("persisted").save()

        -- Delete all of the open buffers
        vim.api.nvim_input("<ESC>:%bd!<CR>")

        -- Don't start saving the session yet
        require("persisted").stop()
      end,
      opts = { pattern = "PersistedTelescopeLoadPre" },
    },
  },
```

### Describe

I want to pin / sticky a buffer through sessions save/switching, more verbose below:

### Reproduce

1. at sessionA
2. :term
3. `<C-\><C-n>` to bring terminal buffer to normal mode (you can now switch to other buffers)
4. open some buffers under sessionA
5. :SessionSave 
6. switch to sessionB --> with above minimal config, terminal buffer from sessionA will be deleted

How to not save terminal buffer from sessionA at step 5? other buffers at step 4 should still be saved to sessionA.

When switch to sessionB at step 6, terminal buffer from sessionA should be avail/merged to sessionB buffers list

**Comment by olimorris** [comment]
Unfortunately this isn't possible with this plugin. It makes use of the inbuilt `:mks` function (`:h mks`) for saving and `source` for loading. Unless there's some way you can restore that buffer via one of the event callbacks?

**Reply by tmpm697** [reply]
for my thinking, i'd need to a label/uid for buffer to exclude it from saving, if this plugins base on builtin `mks` then `mks` should have a way to do this that idk.

**Reply by olimorris** [reply]
No idea unfortunately but I'm more than happy to accept a PR
